amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
My thought is that the test author's constraints should take precedent, but it does make sense that environmental constraints could be appended. My reasoning is that the test author should only be specifying constraints that are required for the deployment, or the specific test being executed - overriding those would seemingly only lead to false negatives.

If you approve of that logic, I can update my branch. I also wonder if we shouldn't change the environment variable name to make the interface's behavior more obvious - e.g. JUJU_TEST_EXTRA_CONSTRAINTS? (or maybe AMULET_EXTRA_CONSTRAINTS? not sure what the best namespace is).